src/qt/paymentserver.cpp

```diff
@@ -770,7 +770,7 @@ void PaymentServer::handlePaymentACK(const QString& paymentACKMsg)
 
 bool PaymentServer::verifyNetwork(const payments::PaymentDetails& requestDetails)
 {
-    bool fVerified = requestDetails.network() == Params().NetworkIDString();
+    bool fVerified = requestDetails.genesis() == Params().hashGenesisBlock;
```

Comment:
qt/paymentserver.cpp: In static member function ‘static bool PaymentServer::verifyNetwork(const payments::PaymentDetails&)’:
qt/paymentserver.cpp:773:59: error: ‘const class CChainParams’ has no member named ‘HashGenesisBlock’
     bool fVerified = requestDetails.genesis() == Params().HashGenesisBlock();
                                                           ^
qt/paymentserver.cpp:777:56: error: ‘const class payments::PaymentDetails’ has no member named ‘HashGenesisBlock’
             .arg(QString::fromStdString(requestDetails.HashGenesisBlock()))

Going to go with "Looks like no"